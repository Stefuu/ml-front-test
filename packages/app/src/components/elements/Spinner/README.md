# Spinner

## Element used to signal loading items
