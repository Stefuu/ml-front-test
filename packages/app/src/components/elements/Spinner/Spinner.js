import styled from 'styled-components'

const Spinner = styled.div`
  animation: rotate 1s infinite linear;
  border: solid ${props => props.thickness}px transparent;
  border-radius: 50%;
  border-right-color: ${props => props.color};
  border-top-color: ${props => props.color};
  display: inline-block;
  height: ${props => props.size}px;
  width: ${props => props.size}px;

  @keyframes rotate {
    from { transform: rotate(0deg) }
    to { transform: rotate(359deg) }
  }
`

Spinner.defaultProps = {
  color: 'rgba(255,255,255,.8)',
  size: 20,
  thickness: 3
}

export default Spinner
