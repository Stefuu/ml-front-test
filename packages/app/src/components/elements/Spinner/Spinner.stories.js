import React from 'react'
import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered'
import { withKnobs, color, number } from '@storybook/addon-knobs'
import { doc } from 'storybook-readme'
import Spinner from './Spinner'
import readmeFile from './README.md'

storiesOf('app/components/icons/Spinner', module)
  .add('README', doc(readmeFile))
  .addDecorator(centered)
  .addDecorator(withKnobs)
  .addParameters({ jest: ['Spinner.test.js'] })
  .add('simple', () => (
    <Spinner size={number('size', 20)} thickness={number('thickness', 3)} color={color('color', '#000')} />
  ))
