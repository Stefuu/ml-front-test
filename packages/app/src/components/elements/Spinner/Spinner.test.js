import React from 'react'
import { render } from 'react-testing-library'
import Spinner from './Spinner'

describe('Spinner', () => {
  it('should render [snapshot]', () => {
    const { container } = render(<Spinner />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
