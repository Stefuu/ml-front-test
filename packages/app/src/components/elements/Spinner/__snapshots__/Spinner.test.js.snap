// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render [snapshot] 1`] = `
<div
  class="sc-bdVaJa hvYkfp"
  color="rgba(255,255,255,.8)"
  size="20"
/>
`;
